File: easy/Projection Area of 3D Shapes/Go/main.go
// Problem: Projection Area of 3D Shapes
// Link to the problem: https://leetcode.com/problems/projection-area-of-3d-shapes/
func projectionArea(grid [][]int) int {
	var n int = len(grid)
	var ans int = 0
	for i := 0; i < n; i++ {
		var a int = 0
		var b int = 0
		for j := 0; j < n; j++ {
			a = max(a, grid[i][j])
			b = max(b, grid[j][i])
			if grid[i][j] > 0 {
				ans++
			}
		}
		ans += (a + b)
	}
	return ans
}

File: easy/Projection Area of 3D Shapes/Rust/main.rs
// Problem: Projection Area of 3D Shapes
// Link to the problem: https://leetcode.com/problems/projection-area-of-3d-shapes/
impl Solution {
    pub fn projection_area(grid: Vec<Vec<i32>>) -> i32 {
        let n: usize = grid.len();
        let mut ans: i32 = 0;
        for i in 0..n {
            let mut a: i32 = 0;
            let mut b: i32 = 0;
            for j in 0..n {
                a = a.max(grid[i][j]);
                b = b.max(grid[j][i]);
                if (grid[i][j] > 0) {
                    ans += 1;
                }
            }
            ans += (a + b);
        }
        return ans;
    }
}
